src/partial.ts

```diff
@@ -0,0 +1,76 @@
+import { purry } from './purry';
+
+type RemovePrefix<
+  T extends ReadonlyArray<unknown>,
+  U extends ReadonlyArray<unknown>,
+> = T extends [...U, ...infer V] ? V : never;
+
+/**
+ * Creates a function that calls `func` with `args` put before the arguments
+ * it receives.
+ *
+ * **Note:** This method doesn't set the `length` property of partially applied
+ * functions.
+ *
+ * @param func the function to wrap
+ * @param args the arguments to put before
+ * @returns a partially applied function
+ *
+ * @signature
+ *    R.partial(args, func)
+ *
+ * @example
+ *    const fn = (x, y, z) => `${x}, ${y}, and ${z}`
+ *    const partialFn = R.partial([1, 2], fn)
+ *    partialFn(3) // => 1, 2, and 3
+ *
+ * @dataFirst
+ * @category Function
+ * @see partialRight
+ */
+export function partial<
+  A extends ReadonlyArray<unknown>,
+  B extends ReadonlyArray<unknown>,
+  C,
+>(args: [...A], func: (...args: B) => C): (...rest: RemovePrefix<B, A>) => C;
```

Comment:
There's a slight issue with how this is implemented now, in regards to when the "A" type is not a prefix of the arguments for `func`. The invocation of "partial" would be successful and won't result in any error on this line, but the returned function would take arguments as never, e.g. `foo(...args: never)` so only when that function is called will the user see the issue. This could be hard to debug if the function is called much later or further downstream from the partial call when a lot of typescript inference is relied on.

One trick to provide more context is to use a literal string instead of `never` for the `RemovePreifx` type, e.g. 

```ts
type RemovePrefix<
  T extends ReadonlyArray<unknown>,
  U extends ReadonlyArray<unknown>,
> = T extends [...U, ...infer V] ? V : "ERROR(partial): the provided argument types are not a prefix of the function's expected argument types";
```

Reply:
hm, i like this, but in this case i had to put the literal string in an array to make it a rest argument type. idk if there's a better solution here